tinygrad/renderer/assembly.py

```diff
@@ -165,16 +165,17 @@ def _cast(a, dtype:DType, atype:DType, bitcast=False, u=None, pred=False):
         elif uop is UOps.DEFINE_ACC:
           if dtype.count > 1:
             r[u] = [ssa('acc', dtype=self.types[dtype.scalar()]) for _ in range(dtype.count)]
-            for uu in r[u]: kk(f"mov.b{self.types[dtype.scalar()][1:]} {uu}, {const(src[0].arg, dtype.scalar())};")
+            for uu in r[u]: kk(f"mov.b{self.types[dtype.scalar()][1:]} {uu}, {const(src[0].src[0].arg, dtype.scalar())};")
           else: kk(f"mov.{f'b{self.types[dtype][1:]}' if dtype != dtypes.bool else 'pred'} {ssa('acc', u)}, {const(src[0].arg, dtype)};")
         elif uop is UOps.SPECIAL:
           assert args[1][0] != "i", "idx not supported"
           kk(f"mov.u32 %{args[1]}, {(self.gid if args[1][0] == 'g' else self.lid)[args[0]]};")
           r[u] = "%" + args[1]
           kernel = [f".reg .u32 %{args[1]};"] + kernel
         elif uop is UOps.CONST:
-          if dtype.count > 1: r[u] = [const(args, dtype.scalar(), mov=True) for _ in range(dtype.count)]
-          else: r[u] = const(args, dtype, mov=True)
+          assert dtype.count == 1
+          n = next((x.dtype.count for x in uops.uops if x.op == UOps.VECTORIZE and all(s is u for s in x.src)), None) # type: ignore
+          r[u] = [const(args, dtype, mov=True) for _ in range(n)] if n else const(args, dtype, mov=True)
```

Comment:
This probably deserves some explanation -- I found that the process_replay for CONST is sensitive to the topo sort order of the uopgraph because the CONST render kk's a mov instruction. When VECTORIZE'd CONST replaces CONST, the sorting changes in a way that makes it difficult to maintain process_replay parity. There were two main options I considered:
1. remove the type_verify assert for CONST and add a pat in this renderer that rewrites VECTORIZE'd CONST back to the original implicitly vectorized CONST, then push another PR that adds the assert back, changes the renderer, and introduces a process_replay diff.
2. in the CONST render, search the graph for the child VECTORIZE'd CONST and render it.

I chose the second because the first option seems against the spirit of the PR, and I figure the second can be reverted in a subsequent PR that introduces a process_replay diff.

Reply:
This actually doesn't work because there are now situations where CONST is shared by VECTORIZE and another UOp that expects a scalar CONST, so this generates bogus instructions, hence the CI failures. I'm now thinking it may not be possible to maintain process_replay parity in the PTX backend without the WAR I mentioned above, or pushing a different PR that makes the CONST renders insensitive to the sort order.